### ISSUE TITLE
Install package at the same time (if needed)

### ISSUE THREAD

**Issue by LuckyTurtleDev** [Jguer/yay#1449]
### Is your feature request related to a problem? Please describe.
Package A depend an special version of another package B. If package A will be updated and depend on a newer version of package B. Yay could not update this package A, because it tries to update Package B first, which breaks the dependency of the old Package A.

### Describe the solution you'd like
Package A and B should be installed at the same time and not behind each other.

### Describe alternatives you've considered
Manuel uninstall Package B before updating package A.

### Additional context
Furthermore I think it is much more pleasant to install all package (executed dependency) at once after compiling all packages, instead of a user interaction to enter the password every time a package is ready.

**Comment by Jguer** [comment]
Hey @Lukas1818 , have you tried using `--batchinstall`?

**Comment by LuckyTurtleDev** [comment]
No I did not.
It does exactly make, what I would like to do. Thanks
Is it possible to set `--batchinstall` as default?

**Comment by LuckyTurtleDev** [comment]
But is still think yay should automatic install, both package at same time, if it run in a deadlock as described above.

**Comment by Morganamilo** [comment]
It's off by default because stuff builds against libraries that are installed on the system. I imagine it's possible to update program foo and libbar. If libbar is built but not installed then, foo would build against the old version.

**Comment by LuckyTurtleDev** [comment]
I mean yay should only install the two package at the same time, if it run in the deadlock. If their no deadlock, yay could install the packages behind each other.

**Comment by Morganamilo** [comment]
Yes that's the idea, but this is why batchinstall is off by default.

**Comment by LuckyTurtleDev** [comment]
It is possible that yay could automatic detect this and automatic install the package at the same time, if it needed?

**Comment by no1xsyzy** [comment]
I just ran into same issue between `zfs-dkms` and `zfs-utils`, in which case uninstalling `zfs-dkms` (the dependent) is not a wise choice, since it will tweak around the kernel and need to do a time-consuming `dkms install`.

And `--batchinstall` might still be a problem if batch install and serial install should be combined in this update.

Question: Will yay continue to upgrade other packages after one package upgrade failed, as long as they are not depending on the failing package?

**Comment by brettinternet** [comment]
batchinstall doesn't work for me.

```
$ yay -S zfs-utils zfs-dkms
...
error: failed to prepare transaction (could not satisfy dependencies)
:: installing zfs-utils (2.1.2-1) breaks dependency 'zfs-utils=2.1.1' required by zfs-dkms
 -> exit status 1
```

```
$ yay --show -g
{
  ...
  "batchinstall": true
}
```

Should it work?

**Comment by brettinternet** [comment]
bump for stalebot

**Comment by floitsch** [comment]
My package isn't yet popular enough to be in 'extra', but would `--batchinstall` also cover the case where one package is in extra, and one is on AUR?

**Comment by brauliobo** [comment]
`--batchinstall` doesn't help here, I'm getting:
```
:: installing nginx (1.26.2-1) breaks dependency 'nginx=1.26.1' required by nginx-mod-rtmp`
```

It should update `nginx-mod-rtmp` from AUR together with `nginx`, when I try an update with
```
nice yay -Syu --noconfirm --batchinstall
```
Also the same error with:
```
nice yay -S nginx nginx-mod-rtmp --batchinstall
```
The workaround
```
yay -R nginx-mod-rtmp; yay -S nginx; yay -S nginx-mod-rtmp --noconfirm
```

**Comment by hockeymikey** [comment]
I'm running into this too with oracle-instantclient-basic and the rest of the packages that rely on it (oracle-instantclient-sqlplus, etc)
